test/test_helpers.py

```diff
@@ -130,3 +132,19 @@ def create_model(self):
         meta = CharmMeta()
         model = Model(meta, backend)
         return model
+
+
+class TestYAML(unittest.TestCase):
+    def test_safe_load(self):
+        d = _yaml.safe_load('foo: bar\nbaz: 123\n')
+        self.assertEqual(len(d), 2)
+        self.assertEqual(d['foo'], 'bar')
+        self.assertEqual(d['baz'], 123)
+
+    def test_safe_dump(self):
+        s = _yaml.safe_dump({'foo': 'bar', 'baz': 123})
+        self.assertEqual(s, 'baz: 123\nfoo: bar\n')
+
+        f = io.StringIO()
+        s = _yaml.safe_dump({'foo': 'bar', 'baz': 123}, stream=f)
+        self.assertEqual(f.getvalue(), 'baz: 123\nfoo: bar\n')
```

Comment:
The only test I would add is that the load and dump are genuinely safe and refuse non basic content. Serializing a class should should sufficient to trigger an exception. I don't remember the exact yaml syntax but you can serialize with regular unsafe dump to get the content that would be unsafe to load.

Reply:
Agreed, thanks -- done.